/*
 * Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package privacy_sandbox.bidding_auction_servers.bidding_service;

import "services/common/blob_fetch/fetch_mode.proto";

message BuyerCodeFetchConfig {
   // The js generateBid script.
   string bidding_js_path = 1;

   // URL endpoint for fetching AdTech code blob, js file.
   string bidding_js_url = 2;

   // URL endpoint to the wasm file. AdTechs can choose to include wasm file
   // which will be combined with the js file before loading into Roma.
   string bidding_wasm_helper_url = 3;

   // Period of how often to fetch AdTech code blob from the URL endpoint.
   int64 url_fetch_period_ms = 4;

   // A time out limit for HttpsFetcherAsync client to stop executing FetchUrl.
   int64 url_fetch_timeout_ms = 5;

   // Allow buyer debug URL generation.
   bool enable_buyer_debug_url_generation = 6;

   // Allow handling of console.logs from AdTech script execution.
   bool enable_adtech_code_logging = 8;

   // URL endpoint for fetching AdTech code blob, js file for protected app
   // signals.
   string protected_app_signals_bidding_js_url = 9;

   // URL endpoint to the wasm file. AdTechs can choose to include wasm file
   // which will be combined with the js file before loading into Roma.
   string protected_app_signals_bidding_wasm_helper_url = 10;

   // URL endpoint for fetching AdTech code blob, js file.
   string prepare_data_for_ads_retrieval_js_url = 11;

   // URL endpoint to the wasm file. AdTechs can choose to include wasm file
   // which will be combined with the js file before loading into Roma.
   string prepare_data_for_ads_retrieval_wasm_helper_url = 12;

   // The name of a bucket from which to fetch protected auction js/wasm code
   // blobs. All blobs will be fetched from this bucket.
   // TODO: The concept of a separate wasm_helper blob is not yet supported if
   // using bucket-based fetching.
   string protected_auction_bidding_js_bucket = 14;

   // The name of the bucket's default protected auction js/wasm code blob to
   // use. The default will be used if the bid request does not specify a
   // version.
   string protected_auction_bidding_js_bucket_default_blob = 15;

   // The name of a bucket from which to fetch protected app signal js/wasm code
   // blobs. All blobs will be fetched from this bucket.
   // TODO: The concept of a separate wasm_helper blob is not yet supported if
   // using bucket-based fetching.
   string protected_app_signals_bidding_js_bucket = 16;

   // The name of the bucket's default protected app signal js/wasm code blob to
   // use. The default will be used if the bid request does not specify a
   // version.
   string protected_app_signals_bidding_js_bucket_default_blob = 17;

   // The name of a bucket from which to fetch prepareDataForAdsRetrieval
   // js/wasm code blobs. All blobs will be fetched from this bucket.
   // TODO: The concept of a separate wasm_helper blob is not yet supported if
   // using bucket-based fetching.
   string ads_retrieval_js_bucket = 18;

   // The name of the bucket's default prepareDataForAdsRetrieval js/wasm code
   // blob to use. The default will be used if the bid request does not specify
   // a version.
   string ads_retrieval_bucket_default_blob = 19;

   blob_fetch.FetchMode fetch_mode = 20;

   // Enables private aggregate reporting.
   bool enable_private_aggregate_reporting = 21;

   // The executable generateBid script.
   string bidding_executable_path = 22;

   // URL endpoint for fetching AdTech code blob, executable file.
   string bidding_executable_url = 23;

   // The name of a bucket from which to fetch protected auction executable code
   // blobs. All blobs will be fetched from this bucket.
   string protected_auction_bidding_executable_bucket = 24;

   // The name of the bucket's default protected auction executable code blob to
   // use. The default will be used if the bid request does not specify a
   // version.
   string protected_auction_bidding_executable_bucket_default_blob = 25;
}
